server/weaverbird/backends/sql_translator/steps/custom.py

```diff
@@ -0,0 +1,52 @@
+from distutils import log
+
+from weaverbird.backends.sql_translator.steps.utils.query_transformation import (
+    build_selection_query,
+)
+from weaverbird.backends.sql_translator.types import (
+    SQLPipelineTranslator,
+    SQLQuery,
+    SQLQueryDescriber,
+    SQLQueryExecutor,
+    SQLQueryRetriever,
+)
+from weaverbird.pipeline.steps import CustomStep
+
+
+def translate_custom(
+    step: CustomStep,
+    query: SQLQuery,
+    index: int,
+    sql_query_retriever: SQLQueryRetriever = None,
+    sql_query_describer: SQLQueryDescriber = None,
+    sql_query_executor: SQLQueryExecutor = None,
+    sql_translate_pipeline: SQLPipelineTranslator = None,
+    subcall_from_other_pipeline_count: int = None,
+) -> SQLQuery:
+    query_name = f'CUSTOM_STEP_{index}'
+
+    log.debug(
+        '############################################################'
+        f'query_name: {query_name}\n'
+        '------------------------------------------------------------'
+        f'step: {step}\n'
+        f'query.transformed_query: {query.transformed_query}\n'
+        f'query.metadata_manager.query_metadata: {query.metadata_manager.retrieve_query_metadata()}\n'
+    )
+    injected_query = step.query.replace('##PREVIOUS_STEP##', query.query_name).replace(';', '')
```

Comment:
- Maybe we should make a trim for extras spaces in this incoming query
- Maybe we should check if the length of the query is upper than a minimal value before proceed it, otherwise, we just replay the precedent step ? 🤔 
- We should add a validator here i think... even if it's not a perfect one... (for the snowflake synthax, this is a [source code](https://gist.github.com/tamiroze/dc0bdd596ed2c6b70fe921061401e739) for converting normal sql to snowflake sql... maybe we can take some regex here to add to our validator ? 🤔  )

Reply:
1. Yes, good idea I'll sanitize more 
2. This check is performed by the frontend's validation
3. I Guess we'll handle that by an enhanced frontend validation